In [ ]:
import pandas as pd

In [ ]:
df=pd.read_csv('/content/StudentsPerformance.csv')
df

gender race/ethnicity parental level of education         lunch  \
0    female        group B           bachelor's degree      standard   
1    female        group C                some college      standard   
2    female        group B             master's degree      standard   
3      male        group A          associate's degree  free/reduced   
4      male        group C                some college      standard   
..      ...            ...                         ...           ...   
995  female        group E             master's degree      standard   
996    male        group C                 high school  free/reduced   
997  female        group C                 high school  free/reduced   
998  female        group D                some college      standard   
999  female        group D                some college  free/reduced   

    test preparation course  math score  reading score  writing score  
0                      none          72             72             74  
1                 completed          69             90             88  
2                      none          90             95             93  
3                      none          47             57             44  
4                      none          76             78             75  
..                      ...         ...            ...            ...  
995               completed          88             99             95  
996                    none          62             55             55  
997               completed          59             71             65  
998               completed          68             78             77  
999                    none          77             86             86  

[1000 rows x 8 columns]

In [ ]:
df[['math score','reading score','writing score']].mean()

math score       66.089
reading score    69.169
writing score    68.054
dtype: float64

In [ ]:
df[['math score','reading score','writing score']].median()

math score       66.0
reading score    70.0
writing score    69.0
dtype: float64

In [ ]:
df[['math score','reading score','writing score']].mode()

math score  reading score  writing score
0          65             72             74

distribution symétrique car moyenne=mediane

# **Exercice 2**

In [ ]:
# a
etendu=df[['math score','reading score','writing score']].max()-df[['math score','reading score','writing score']].min()
etendu

math score       100
reading score     83
writing score     90
dtype: int64

interpretation : etendu élevée



In [ ]:
#b
écrat_type=df[['math score','reading score','writing score']].std()
variante=df[['math score','reading score','writing score']].var()
écrat_type




math score       15.163080
reading score    14.600192
writing score    15.195657
dtype: float64

In [ ]:
variante

math score       229.918998
reading score    213.165605
writing score    230.907992
dtype: float64

In [ ]:
écrat_type.idxmax()


'writing score'

math score	15.163080 forte variabilité


reading score	14.600192 faible variabilité


writing score	15.195657 forte variabilité


# **Exercice 3**

In [ ]:
# a
Q1=df[['math score','reading score','writing score']].quantile(0.25)
Q3=df[['math score','reading score','writing score']].quantile(0.75)
IQR=Q3-Q1

In [ ]:
Q1

math score       57.00
reading score    59.00
writing score    57.75
Name: 0.25, dtype: float64

In [ ]:
Q3

math score       77.0
reading score    79.0
writing score    79.0
Name: 0.75, dtype: float64

In [ ]:
IQR

math score       20.00
reading score    20.00
writing score    21.25
dtype: float64

In [ ]:
#b Détection des Valeurs Aberrantes
Limite_inférieure = Q1 - 1.5 * IQR
Limite_supérieure = Q3 + 1.5 * IQR

In [ ]:
Limite_inférieure

math score       27.000
reading score    29.000
writing score    25.875
dtype: float64

In [ ]:
Limite_supérieure

math score       107.000
reading score    109.000
writing score    110.875
dtype: float64

# **Exercice 4**

In [ ]:
import plotly.express as px

In [ ]:
#a) Graphiques
fig_math = px.histogram(df, x='math score',
                        nbins=10,
                        title='Histogramme - Math Score')
fig_math.show()

fig_reading = px.histogram(df, x='reading score',
                           nbins=10,
                           title='Histogramme - Reading Score')
fig_reading.show()

fig_writing = px.histogram(df, x='writing score',
                           nbins=10,
                           title='Histogramme - Writing Score')
fig_writing.show()

In [ ]:
#Boîtes à moustaches (Box plots) → Quartiles et valeurs aberrantes

fig_box = px.box(df,
                 y=['math score', 'reading score', 'writing score'],
                 title='Box Plot des Scores')
fig_box.show()


In [ ]:
#Graphiques en barres → Comparaisons entre matières
means = df[['math score', 'reading score', 'writing score']].mean().reset_index()
means.columns = ['Subject', 'Mean Score']

fig_bar = px.bar(means,
                 x='Subject',
                 y='Mean Score',
                 title='Comparaison des Moyennes')
fig_bar.show()

# **EXERCICE 5**

In [ ]:
#a
correlation = df[['math score', 'reading score', 'writing score']].corr()

fig_corr = px.imshow(correlation,
                     text_auto=True,
                     title="Heatmap de Corrélation")
fig_corr.show()


In [ ]:
#b
fig_scatter = px.scatter(df,
                         x='math score',
                         y='reading score',
                         title='Math vs Reading',
                         trendline='ols')
fig_scatter.show()

In [ ]:
fig_matrix = px.scatter_matrix(df,
              dimensions=['math score', 'reading score', 'writing score']
,
                                title='Scatter Matrix des Scores')
fig_matrix.show()


In [ ]:
fig_prep = px.box(df,
                  x='test preparation course',
                  y='math score',
                  title='Impact du Test Preparation Course')
fig_prep.show()


In [ ]:
fig_lunch = px.box(df,
                   x='lunch',
                   y='math score',
                   title='Impact du Lunch')
fig_lunch.show()


In [ ]:
fig_parent = px.box(df,
                    x='parental level of education',
                    y='math score',
                    title="Impact de l'Éducation Parentale")
fig_parent.update_layout(xaxis_tickangle=45)
fig_parent.show()